src/sentry_core.c

```diff
@@ -125,11 +150,10 @@ sentry_shutdown(void)
 {
     sentry_end_session();
 
-    sentry__mutex_lock(&g_options_mutex);
-    sentry_options_t *options = g_options;
-    sentry__mutex_unlock(&g_options_mutex);
-
-    if (options) {
+    SENTRY_WITH_OPTIONS (options) {
+        // we unset the global here, to try to minimize the chance of errors
+        // with unsynchronized concurrent accesses
+        g_options = NULL;
```

Comment:
This lock scope can live a bit shorter. After setting `g_options` to NULL, remember the old reference and the exit the scope.